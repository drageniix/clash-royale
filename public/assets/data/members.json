{"members":[{"tag":"#PYP8VCRG","playername":"Goofyjay2","trophies":4828,"clanrank":1,"role":"elder","donations":408,"donationsreceived":240,"wars":"15","battles":"15","wins":"9","losses":"6","missed":"0","winRatio":"60.00","warParticipationRatio":"100.00","donationRatio":"170.00"},{"tag":"#80CYLGVL","playername":"Jurgen","trophies":4821,"clanrank":2,"role":"elder","donations":256,"donationsreceived":200,"wars":"13","battles":"13","wins":"11","losses":"2","missed":"0","winRatio":"84.62","warParticipationRatio":"86.67","donationRatio":"128.00"},{"tag":"#2CCL092UY","playername":"TitoKilo","trophies":4802,"clanrank":3,"role":"elder","donations":84,"donationsreceived":200,"wars":"15","battles":"17","wins":"12","losses":"5","missed":"0","winRatio":"70.59","warParticipationRatio":"100.00","donationRatio":"42.00"},{"tag":"#92G8UPJU","playername":"luc","trophies":4728,"clanrank":4,"role":"elder","donations":105,"donationsreceived":200,"wars":"10","battles":"10","wins":"4","losses":"6","missed":"0","winRatio":"40.00","warParticipationRatio":"66.67","donationRatio":"52.50"},{"tag":"#9LQCCVLY","playername":"Studmuffin","trophies":4724,"clanrank":5,"role":"elder","donations":214,"donationsreceived":200,"wars":"13","battles":"12","wins":"9","losses":"3","missed":"1","winRatio":"75.00","warParticipationRatio":"86.67","donationRatio":"107.00"},{"tag":"#UVCGVVQC","playername":"ARYA","trophies":4676,"clanrank":6,"role":"elder","donations":245,"donationsreceived":200,"wars":"5","battles":"5","wins":"1","losses":"4","missed":"0","winRatio":"20.00","warParticipationRatio":"33.33","donationRatio":"122.50"},{"tag":"#C0G2JQ2P","playername":"Jazz","trophies":4605,"clanrank":7,"role":"elder","donations":290,"donationsreceived":160,"wars":"15","battles":"17","wins":"11","losses":"6","missed":"0","winRatio":"64.71","warParticipationRatio":"100.00","donationRatio":"181.25"},{"tag":"#8QCRQ8RR","playername":"WEL3A","trophies":4525,"clanrank":8,"role":"elder","donations":255,"donationsreceived":180,"wars":"14","battles":"16","wins":"8","losses":"8","missed":"0","winRatio":"50.00","warParticipationRatio":"93.33","donationRatio":"141.67"},{"tag":"#RGLVUQJC","playername":"Nico","trophies":4519,"clanrank":9,"role":"elder","donations":0,"donationsreceived":0,"wars":"12","battles":"12","wins":"9","losses":"3","missed":"0","winRatio":"75.00","warParticipationRatio":"80.00","donationRatio":"NaN"},{"tag":"#29UGCU0LY","playername":"lDrago45","trophies":4504,"clanrank":10,"role":"elder","donations":56,"donationsreceived":120,"wars":"15","battles":"15","wins":"10","losses":"5","missed":"0","winRatio":"66.67","warParticipationRatio":"100.00","donationRatio":"46.67"},{"tag":"#2YGQQ0V2C","playername":"Judah","trophies":4472,"clanrank":11,"role":"elder","donations":367,"donationsreceived":140,"wars":"14","battles":"15","wins":"6","losses":"9","missed":"0","winRatio":"40.00","warParticipationRatio":"93.33","donationRatio":"262.14"},{"tag":"#YQ820CJ2","playername":"JimmyJoe","trophies":4466,"clanrank":12,"role":"elder","donations":152,"donationsreceived":160,"wars":"9","battles":"10","wins":"6","losses":"4","missed":"0","winRatio":"60.00","warParticipationRatio":"60.00","donationRatio":"95.00"},{"tag":"#9VUJVGVU","playername":"El Bar","trophies":4460,"clanrank":13,"role":"elder","donations":270,"donationsreceived":200,"wars":"14","battles":"15","wins":"5","losses":"10","missed":"0","winRatio":"33.33","warParticipationRatio":"93.33","donationRatio":"135.00"},{"tag":"#U9UL9U9U","playername":"Mephobos","trophies":4455,"clanrank":14,"role":"coleader","donations":386,"donationsreceived":200,"wars":"15","battles":"15","wins":"6","losses":"9","missed":"0","winRatio":"40.00","warParticipationRatio":"100.00","donationRatio":"193.00"},{"tag":"#88PU8282","playername":"Dragon King","trophies":4436,"clanrank":15,"role":"member","donations":38,"donationsreceived":160,"wars":"7","battles":"7","wins":"5","losses":"2","missed":"0","winRatio":"71.43","warParticipationRatio":"46.67","donationRatio":"23.75"},{"tag":"#VPVLQJ2V","playername":"Edu","trophies":4432,"clanrank":16,"role":"elder","donations":244,"donationsreceived":200,"wars":"15","battles":"16","wins":"13","losses":"3","missed":"0","winRatio":"81.25","warParticipationRatio":"100.00","donationRatio":"122.00"},{"tag":"#8LVVQRQC","playername":"Sorin Markov","trophies":4415,"clanrank":17,"role":"member","donations":174,"donationsreceived":192,"wars":"13","battles":"12","wins":"6","losses":"6","missed":"1","winRatio":"50.00","warParticipationRatio":"86.67","donationRatio":"90.63"},{"tag":"#V2UCRJ0R","playername":"CAttack","trophies":4411,"clanrank":18,"role":"member","donations":38,"donationsreceived":160,"wars":"5","battles":"4","wins":"2","losses":"2","missed":"1","winRatio":"50.00","warParticipationRatio":"33.33","donationRatio":"23.75"},{"tag":"#8CJP88JLJ","playername":"RAUPP","trophies":4396,"clanrank":19,"role":"elder","donations":208,"donationsreceived":160,"wars":"11","battles":"11","wins":"4","losses":"7","missed":"0","winRatio":"36.36","warParticipationRatio":"73.33","donationRatio":"130.00"},{"tag":"#9JCJLGQL","playername":"Pat3002","trophies":4391,"clanrank":20,"role":"member","donations":124,"donationsreceived":120,"wars":"12","battles":"10","wins":"4","losses":"6","missed":"2","winRatio":"40.00","warParticipationRatio":"80.00","donationRatio":"103.33"},{"tag":"#80UUG9R2U","playername":"Meliaesc","trophies":4376,"clanrank":21,"role":"coleader","donations":386,"donationsreceived":240,"wars":"14","battles":"14","wins":"7","losses":"7","missed":"0","winRatio":"50.00","warParticipationRatio":"93.33","donationRatio":"160.83"},{"tag":"#2LYG9C9R2","playername":"TheThriller","trophies":4366,"clanrank":22,"role":"elder","donations":258,"donationsreceived":160,"wars":"10","battles":"11","wins":"2","losses":"9","missed":"0","winRatio":"18.18","warParticipationRatio":"66.67","donationRatio":"161.25"},{"tag":"#8VJUR0R9Q","playername":"Foxfox105","trophies":4364,"clanrank":23,"role":"elder","donations":458,"donationsreceived":230,"wars":"9","battles":"10","wins":"8","losses":"2","missed":"0","winRatio":"80.00","warParticipationRatio":"60.00","donationRatio":"199.13"},{"tag":"#U922UR8","playername":"ZombifiedOnyx","trophies":4362,"clanrank":24,"role":"elder","donations":44,"donationsreceived":120,"wars":"9","battles":"9","wins":"2","losses":"7","missed":"0","winRatio":"22.22","warParticipationRatio":"60.00","donationRatio":"36.67"},{"tag":"#2PUYVLVCU","playername":"SukonDeezNUTZ","trophies":4340,"clanrank":25,"role":"member","donations":36,"donationsreceived":80,"wars":"12","battles":"12","wins":"7","losses":"5","missed":"1","winRatio":"58.33","warParticipationRatio":"80.00","donationRatio":"45.00"},{"tag":"#9QL0QGLR","playername":"WM","trophies":4331,"clanrank":26,"role":"elder","donations":48,"donationsreceived":120,"wars":"7","battles":"8","wins":"6","losses":"2","missed":"0","winRatio":"75.00","warParticipationRatio":"46.67","donationRatio":"40.00"},{"tag":"#82J0YJ82Q","playername":"Niekotine Frost","trophies":4314,"clanrank":27,"role":"elder","donations":64,"donationsreceived":80,"wars":"12","battles":"11","wins":"5","losses":"6","missed":"1","winRatio":"45.45","warParticipationRatio":"80.00","donationRatio":"80.00"},{"tag":"#8LV909VR","playername":"David_McG_IVB**","trophies":4312,"clanrank":28,"role":"elder","donations":220,"donationsreceived":160,"wars":"13","battles":"13","wins":"3","losses":"10","missed":"0","winRatio":"23.08","warParticipationRatio":"86.67","donationRatio":"137.50"},{"tag":"#22PUP0C","playername":"Kyle","trophies":4243,"clanrank":29,"role":"elder","donations":186,"donationsreceived":160,"wars":"12","battles":"12","wins":"3","losses":"9","missed":"0","winRatio":"25.00","warParticipationRatio":"80.00","donationRatio":"116.25"},{"tag":"#88CGCQ8L","playername":"Viking6514","trophies":4236,"clanrank":30,"role":"elder","donations":262,"donationsreceived":200,"wars":"10","battles":"10","wins":"6","losses":"4","missed":"0","winRatio":"60.00","warParticipationRatio":"66.67","donationRatio":"131.00"},{"tag":"#8LVQ9RJYJ","playername":"MD Hosny","trophies":4224,"clanrank":31,"role":"elder","donations":104,"donationsreceived":160,"wars":"4","battles":"4","wins":"2","losses":"2","missed":"0","winRatio":"50.00","warParticipationRatio":"26.67","donationRatio":"65.00"},{"tag":"#LLL9UJ8","playername":"Felipe","trophies":4215,"clanrank":32,"role":"member","donations":180,"donationsreceived":240,"wars":"1","battles":"1","wins":"1","losses":"0","missed":"0","winRatio":"100.00","warParticipationRatio":"6.67","donationRatio":"75.00"},{"tag":"#2200VCJ9","playername":"Czarofzar","trophies":4214,"clanrank":33,"role":"elder","donations":195,"donationsreceived":200,"wars":"13","battles":"13","wins":"1","losses":"12","missed":"0","winRatio":"7.69","warParticipationRatio":"86.67","donationRatio":"97.50"},{"tag":"#LCRQURRR","playername":"B3ASTB0Y","trophies":4193,"clanrank":34,"role":"member","donations":32,"donationsreceived":160,"wars":"2","battles":"2","wins":"0","losses":"2","missed":"0","winRatio":"0.00","warParticipationRatio":"13.33","donationRatio":"20.00"},{"tag":"#PP2P898R","playername":"Piggy123","trophies":4191,"clanrank":35,"role":"elder","donations":104,"donationsreceived":200,"wars":"10","battles":"10","wins":"8","losses":"2","missed":"0","winRatio":"80.00","warParticipationRatio":"66.67","donationRatio":"52.00"},{"tag":"#28GVQ0G98","playername":"Ryan","trophies":4172,"clanrank":36,"role":"elder","donations":72,"donationsreceived":120,"wars":"10","battles":"10","wins":"6","losses":"4","missed":"0","winRatio":"60.00","warParticipationRatio":"66.67","donationRatio":"60.00"},{"tag":"#8YQC89PV","playername":"Z Titan","trophies":4160,"clanrank":37,"role":"member","donations":36,"donationsreceived":80,"wars":"5","battles":"5","wins":"2","losses":"3","missed":"0","winRatio":"40.00","warParticipationRatio":"33.33","donationRatio":"45.00"},{"tag":"#28PVUV89J","playername":"esteban","trophies":4158,"clanrank":38,"role":"member","donations":80,"donationsreceived":160,"wars":"4","battles":"5","wins":"3","losses":"2","missed":"0","winRatio":"60.00","warParticipationRatio":"26.67","donationRatio":"50.00"},{"tag":"#QPUL2V29","playername":"Soupcan43","trophies":4154,"clanrank":39,"role":"member","donations":60,"donationsreceived":160,"wars":"3","battles":"2","wins":"2","losses":"0","missed":"1","winRatio":"100.00","warParticipationRatio":"20.00","donationRatio":"37.50"},{"tag":"#2CL9PUY8","playername":"Caden#21","trophies":4148,"clanrank":40,"role":"elder","donations":8,"donationsreceived":80,"wars":"7","battles":"7","wins":"3","losses":"4","missed":"0","winRatio":"42.86","warParticipationRatio":"46.67","donationRatio":"10.00"},{"tag":"#922L29UCL","playername":"mikal016","trophies":4088,"clanrank":41,"role":"member","donations":280,"donationsreceived":200,"wars":"8","battles":"8","wins":"2","losses":"6","missed":"0","winRatio":"25.00","warParticipationRatio":"53.33","donationRatio":"140.00"},{"tag":"#2J920Q0LV","playername":"sam23363","trophies":4078,"clanrank":42,"role":"member","donations":192,"donationsreceived":200,"wars":"2","battles":"2","wins":"2","losses":"0","missed":"0","winRatio":"100.00","warParticipationRatio":"13.33","donationRatio":"96.00"},{"tag":"#PULVR2","playername":"awesome","trophies":4075,"clanrank":43,"role":"member","donations":20,"donationsreceived":200,"wars":"14","battles":"15","wins":"7","losses":"8","missed":"0","winRatio":"46.67","warParticipationRatio":"93.33","donationRatio":"10.00"},{"tag":"#8828PPJP","playername":"Reaper","trophies":4059,"clanrank":44,"role":"member","donations":0,"donationsreceived":40,"wars":"0","battles":null,"wins":null,"losses":null,"missed":"0","winRatio":"NaN","warParticipationRatio":"0.00","donationRatio":"0.00"},{"tag":"#9GVPG08G","playername":"ZombifiedTitan","trophies":4048,"clanrank":45,"role":"elder","donations":82,"donationsreceived":160,"wars":"5","battles":"5","wins":"3","losses":"2","missed":"0","winRatio":"60.00","warParticipationRatio":"33.33","donationRatio":"51.25"},{"tag":"#2J0QJG0UV","playername":"so alex","trophies":4048,"clanrank":46,"role":"elder","donations":52,"donationsreceived":160,"wars":"4","battles":"2","wins":"1","losses":"1","missed":"2","winRatio":"50.00","warParticipationRatio":"26.67","donationRatio":"32.50"},{"tag":"#2PLRQUUQQ","playername":"Anubhav","trophies":4026,"clanrank":47,"role":"elder","donations":201,"donationsreceived":120,"wars":"13","battles":"15","wins":"7","losses":"8","missed":"0","winRatio":"46.67","warParticipationRatio":"86.67","donationRatio":"167.50"},{"tag":"#GJLYVL8Y","playername":"richi","trophies":4020,"clanrank":48,"role":"member","donations":58,"donationsreceived":120,"wars":"0","battles":null,"wins":null,"losses":null,"missed":"0","winRatio":"NaN","warParticipationRatio":"0.00","donationRatio":"48.33"},{"tag":"#YCPJJ8G","playername":"lauren","trophies":4008,"clanrank":49,"role":"leader","donations":94,"donationsreceived":120,"wars":"15","battles":"16","wins":"11","losses":"5","missed":"0","winRatio":"68.75","warParticipationRatio":"100.00","donationRatio":"78.33"},{"tag":"#R0UYUCCU","playername":"WhiteCloud","trophies":4000,"clanrank":50,"role":"member","donations":296,"donationsreceived":200,"wars":"1","battles":"1","wins":"0","losses":"1","missed":"0","winRatio":"0.00","warParticipationRatio":"6.67","donationRatio":"148.00"}]}