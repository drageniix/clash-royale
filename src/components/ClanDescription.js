import React from 'react'
import { connect } from 'react-redux'

const ClanDescription = ({admin}) => (
    <section className="description">
        <div className="description__reqs">
            <h2>Clan Requirements</h2>
            <p><strong>Member:</strong> Donate at least {admin.minDonations} cards weekly and participate in {admin.minWarBattles}/10 war battles, missing a maximum of 1 war day battle when you participated in collection day.</p>
            <p><strong>Elder:</strong> Have at least {admin.minTrophies} trophies and donate more than {admin.elderDonations} cards. Must participate in at least {admin.elderWarBattles}/10 wars. Kicking members and sending invitations to those below basic member requirements requires permission.</p>
        </div>
    </section>
)

export default connect(state => ({
    admin: state.admin
}))(ClanDescription)