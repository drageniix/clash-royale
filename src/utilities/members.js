export const capitalizeRole = member =>
    member.role.charAt(0).toUpperCase() + member.role.slice(1);
