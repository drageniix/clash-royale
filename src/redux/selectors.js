import { capitalizeFirstLetter } from '../utilities/basic'

const filterOptions = {
    none: member => member,
    promotion: member => member.eligibleForPromotion,
    probation: member => member.onProbation,
    demotion: member => member.dangerOfDemotion
}

const sortOptions = {
    rank: {
        ascending : (a, b) => a.clanRank - b.clanRank,
        descending : (a, b) => b.clanRank - a.clanRank
    },
    name: {
        ascending: (a, b) => {
            const aName = a.name.toLowerCase()
            const bName = b.name.toLowerCase()
            return aName < bName ? -1 : aName > bName ? 1 : 0
        },
        descending: (a, b) => {
            const aName = a.name.toLowerCase()
            const bName = b.name.toLowerCase()
            return aName < bName ? 1 : aName > bName ? -1 : 0
        }
    },
    donations: {
        ascending: (a, b) => b.donations - a.donations,
        descending: (a, b) => a.donations - b.donations
    },
    warBattltes: {
        ascending: (a, b) => b.warBattles - a.warBattles,
        descending: (a, b) => a.warBattles - b.warBattles
    },
    missed: {
        ascending: (a, b) => b.missedWarBattles - a.missedWarBattles,
        descending: (a, b) =>  a.missedWarBattles - b.missedWarBattles
    },
    role: {
        getValue(role){
            switch(role) {
                case 'leader':
                    return 1
                case 'coleader':
                    return 2
                case 'elder':
                    return 3
                case 'member':
                    return 4
                case 'new':
                    return 5
            }
        },
        ascending(a, b){
            const aRole = this.getValue(a.role)
            const bRole = this.getValue(b.role)

            return aRole - bRole
        },
        descending(a, b){
            const aRole = this.getValue(a.role)
            const bRole = this.getValue(b.role)

            return bRole - aRole
        },
    }
}

export const getMembers = (stateMembers, stateFilter, stateOrder, stateDir) => stateMembers
    .filter(filterOptions[stateFilter])
    .sort(sortOptions[stateOrder][stateDir])
    
export const getSearchResult = (stateQuery, stateMembers) => stateQuery ? stateMembers.find(member => (
    member.name.toLowerCase().includes(stateQuery.toLowerCase().trim()) ||
    member.tag === stateQuery)) : undefined

export const getDirectionIndicator = (stateOrder, stateDir, order) => capitalizeFirstLetter(order) + (stateOrder === order ? (stateDir === 'ascending' ? " ▼" : " ▲") : "")