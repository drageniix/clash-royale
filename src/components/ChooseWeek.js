import React from 'react'
import { setNewWeek } from '../redux/actions'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

export class ChooseWeek extends React.Component {
    setNewWeek = index => () => {
        this.props.setNewWeek(this.props.lastWeeks, index)
    }

    render(){
        return (
            <span className="weekSelection">
                <a>History</a>
                <div className="weekSelection__weeks">
                    {this.props.lastWeeks.map((item, index) => (
                        <li className={(this.props.current === index ? 'selectedWeek' : '')} 
                            key={index} 
                            onClick={this.setNewWeek(index)}>
                            {item.display}</li>
                    ))}
                </div>
            </span>
        )
    }

} 

const mapStateToProps = state => ({
    current: state.current,
    lastWeeks: state.api.lastWeeks
})

const mapDispatchToProps = dispatch => ({
    setNewWeek: (lastWeeks, current) => dispatch(setNewWeek(lastWeeks, current))
})

ChooseWeek.propTypes = {
    setNewWeek: PropTypes.object.isRequired,
    current: PropTypes.number.isRequired,
    lastWeeks: PropTypes.array.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(ChooseWeek)