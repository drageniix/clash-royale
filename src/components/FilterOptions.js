import React from 'react';
import { getSelectedFilterClass } from '../redux/selectors';
import { setFilter } from '../redux/actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class FilterOptions extends React.Component {
    onFilter = filterType => () => {
        this.props.onFilter(filterType);
    };

    render() {
        return (
            <div className="filter">
                <div className="filter__vaporware" />
                <ul className="filter__list">
                    <li
                        className={this.props.getSelectedFilterClass('none')}
                        onClick={this.onFilter('byNone')}
                    >
                        All Members
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'promotion'
                        )}
                        onClick={this.onFilter('byPromotion')}
                    >
                        Promotion
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'probation'
                        )}
                        onClick={this.onFilter('byProbation')}
                    >
                        Probation
                    </li>
                    <li
                        className={this.props.getSelectedFilterClass(
                            'demotion'
                        )}
                        onClick={this.onFilter('byDemotion')}
                    >
                        Demotion
                    </li>
                </ul>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    getSelectedFilterClass: filter =>
        getSelectedFilterClass(state.filters.filter, filter)
});

const mapDispatchToProps = dispatch => ({
    onFilter: filterType => dispatch(setFilter[filterType]())
});

FilterOptions.propTypes = {
    getSelectedFilterClass: PropTypes.func.isRequired,
    onFilter: PropTypes.func.isRequired
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(FilterOptions);
