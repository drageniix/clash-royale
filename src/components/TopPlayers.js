import React from 'react'
import { connect } from 'react-redux';
import { getMembers } from '../redux/selectors'
import { setQuery } from '../redux/actions'

export class TopPlayers extends React.Component {
    onClick = memberTag => e => {
        this.props.onClick(memberTag)
    }

    render() {
        return (
            <div className="topPlayers__type">
                <h4>{this.props.title}</h4>
                <table>
                    <tbody>
                    {this.props.members.map((member, index) => (
                        <tr key={index}>
                            <td>{index + 1}.</td>
                            <td className='pointer member-link' onClick={this.onClick(member.tag)}>
                                {member.name}
                            </td>
                        </tr>
                    ))}
                    </tbody>
                </table>
            </div>
        )
    }
}

const mapStateToProps = (state, { order }) => ({
    members: getMembers(state.api.members, {order}).slice(0, 3),
})

const mapDispatchToProps = dispatch => ({
    onClick: query => dispatch(setQuery(query))
})

export default connect(mapStateToProps, mapDispatchToProps)(TopPlayers)