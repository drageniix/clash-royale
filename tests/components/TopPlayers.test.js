import React from 'react'
import members from '../resources/sampleMembers'
import { TopPlayers } from '../../src/components/TopPlayers'
import { getMembers } from '../../src/redux/selectors'

test('render: Description', () => {
    const dispatchSpy = jest.fn()
    const wrapper = shallow(
        <TopPlayers 
            onClick={dispatchSpy}
            members={getMembers(members, { order: 'wins' }).slice(0, 3)}
            title="Most Wins" />
    )

    wrapper.find('.member-link').at(0).simulate('click', {
        preventDefault: () => { }
    })

    expect(dispatchSpy).toHaveBeenCalledWith("#80CYLGVL")
    expect(wrapper).toMatchSnapshot()
})