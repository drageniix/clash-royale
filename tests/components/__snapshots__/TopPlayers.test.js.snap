// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render: Description 1`] = `
<div
  className="topPlayers__type"
>
  <h4>
    Most Wins
  </h4>
  <table>
    <tbody>
      <tr
        key="0"
      >
        <td>
          1
          .
        </td>
        <td
          className="pointer member-link"
          onClick={[Function]}
        >
          Jurgen
        </td>
      </tr>
      <tr
        key="1"
      >
        <td>
          2
          .
        </td>
        <td
          className="pointer member-link"
          onClick={[Function]}
        >
          Meliaesc
        </td>
      </tr>
      <tr
        key="2"
      >
        <td>
          3
          .
        </td>
        <td
          className="pointer member-link"
          onClick={[Function]}
        >
          lDrago45
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
